src/opts.h

```diff
@@ -16,12 +16,22 @@
 
 #include "natsp.h"
 
+static inline void natsOptions_lock(natsOptions *opts)
+{
+    natsMutex_Lock(opts->mu);
+}
+
+static inline void natsOptions_unlock(natsOptions *opts)
+{
+    natsMutex_Unlock(opts->mu);
+}
+
 #define LOCK_AND_CHECK_OPTIONS(o, c) \
     if (((o) == NULL) || ((c))) \
         return nats_setDefaultError(NATS_INVALID_ARG); \
-    natsMutex_Lock((o)->mu);
+    natsOptions_lock(o);
```

Comment:
You need to undo this change or make a change to `sopts.c` because it fails for Stan options.